rskj-core/src/test/java/co/rsk/peg/union/UnionBridgeIT.java

```diff
@@ -0,0 +1,1101 @@
+            .withSignatureCache(signatureCache)
+            .withUnionBridgeSupport(unionBridgeSupport)
+            .withRepository(repository)
+            .withProvider(bridgeStorageProvider)
+            .withEventLogger(eventLogger)
+            .withBridgeConstants(bridgeMainNetConstants)
+            .withActivations(lovellActivations.forBlock(0))
+            .build();
+
+        bridge = bridgeBuilder
+            .signatureCache(signatureCache)
+            .activationConfig(lovellActivations)
+            .bridgeSupport(bridgeSupport)
+            .transaction(rskTx)
+            .build();
+    }
+
+    private void setupForAllActivations() {
+        BridgeStorageProvider bridgeStorageProvider = new BridgeStorageProvider(
+            repository,
+            BRIDGE_ADDR,
+            bridgeMainNetConstants.getBtcParams(),
+            allActivations.forBlock(0)
+        );
+
+        BridgeEventLogger eventLogger = new BridgeEventLoggerImpl(
+            bridgeMainNetConstants,
+            allActivations.forBlock(0),
+            logs
+        );
+
+        UnionBridgeSupport unionBridgeSupport = unionBridgeSupportBuilder
+            .withEventLogger(eventLogger)
+            .build();
+
+        bridgeSupport = bridgeSupportBuilder
+            .withProvider(bridgeStorageProvider)
+            .withActivations(allActivations.forBlock(0))
+            .withEventLogger(eventLogger)
+            .withUnionBridgeSupport(unionBridgeSupport)
+            .build();
+        bridge = bridgeBuilder
+            .activationConfig(allActivations)
+            .bridgeSupport(bridgeSupport)
+            .build();
+    }
+
+    @Test
+    @Order(0)
+    void setUnionBridgeContractAddressForTestnet_whenLovell_shouldFail() {
+        Function function = SET_UNION_BRIDGE_CONTRACT_ADDRESS_FOR_TESTNET.getFunction();
+        byte[] setUnionBridgeContractAddressData = function.encode(
+            NEW_UNION_BRIDGE_CONTRACT_ADDRESS.toHexString());
+        executeBridgeMethodAndAssertFail(setUnionBridgeContractAddressData);
+        assertNoEventWasEmitted();
+    }
+
+    private void executeBridgeMethodAndAssertFail(byte[] bridgeMethod) {
+        VMException actualException = assertThrows(VMException.class,
+            () -> bridge.execute(bridgeMethod));
+        assertEquals(BridgeIllegalArgumentException.class, actualException.getCause().getClass());
+    }
+
+    @Test
+    @Order(1)
+    void getUnionBridgeContractAddress_whenLovell_shouldFail() {
+        Function function = GET_UNION_BRIDGE_CONTRACT_ADDRESS.getFunction();
+        byte[] getUnionBridgeContractAddressData = function.encode();
+        executeBridgeMethodAndAssertFail(getUnionBridgeContractAddressData);
+        assertNoEventWasEmitted();
+    }
+
+    @Test
+    @Order(2)
+    void getUnionBridgeLockingCap_whenLovell_shouldFail() {
+        Function function = GET_UNION_BRIDGE_LOCKING_CAP.getFunction();
+        byte[] getUnionBridgeLockingCapData = function.encode();
```

Comment:
```suggestion
        byte[] unionBridgeLockingCapData = function.encode();
```
same here and the others